### PR TITLE
Add MessagePort stream example

### PR DESCRIPTION
**Pull request by jasnell**

### CODE REVIEW

File: examples/stream/stream.mjs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  _write (buf, _, cb) {
+    this[kPort].postMessage(buf, [buf.buffer]);
+    cb();
+  }
+
+  _writev (data, cb) {
+    for (let n = 0; n < data.length; n++) {
+      this[kPort].postMessage(data[n].chunk, [data[n].chunk.buffer]);
```
Comment (mcollina) [inline examples/stream/stream.mjs]: I'd recommend to add batch support and pass the array through as-is.

File: examples/stream/stream.mjs [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import {
+  Writable,
+  Readable,
+  Duplex
+} from 'stream';
+
```
Comment (mcollina) [inline examples/stream/stream.mjs:6]: All this files seems good utility that could live in a module.

File: examples/stream/stream.mjs (hunk 1 above)
Comment (jasnell) [reply examples/stream/stream.mjs]: Heh, yeah, I added that to the duplex, forgot to update the Writable.

File: examples/stream/stream.mjs (hunk 2 above)
Comment (jasnell) [reply examples/stream/stream.mjs:6]: Yeah, I'll eventually make it a separate module. Want to harden it up a bit more and work on some benchmarking first.

**Comment by jasnell** [comment]
Btw, @mcollina @addaleax ... in the original version of this I had the worker using an async iterator to read the data off the inbound stream but for some reason it would not close/destroy properly. The _destroy() would call on the `MessagePort` to close but the close callback wouldn't fire. Using the streams API version doesn't have the same issue.

**Pull request merged**